packages/mui-joy/src/styles/defaultTheme.ts

```diff
@@ -61,142 +58,246 @@ export interface FontSize {
   // add string to support css variable value.
-  light: React.CSSProperties['fontWeight'] | string;
-  regular: React.CSSProperties['fontWeight'] | string;
-  medium: React.CSSProperties['fontWeight'] | string;
-  semiBold: React.CSSProperties['fontWeight'] | string;
-  bold: React.CSSProperties['fontWeight'] | string;
-  extraBold: React.CSSProperties['fontWeight'] | string;
-  black: React.CSSProperties['fontWeight'] | string;
+  default: React.CSSProperties['fontWeight'] | string;
+  xs: React.CSSProperties['fontWeight'] | string;
+  sm: React.CSSProperties['fontWeight'] | string;
+  md: React.CSSProperties['fontWeight'] | string;
+  lg: React.CSSProperties['fontWeight'] | string;
+  xl: React.CSSProperties['fontWeight'] | string;
 }
 
 export interface LineHeight {
-  xs: React.CSSProperties['lineHeight'];
+  default: React.CSSProperties['lineHeight'];
   sm: React.CSSProperties['lineHeight'];
-  normal: React.CSSProperties['lineHeight'];
+  md: React.CSSProperties['lineHeight'];
   lg: React.CSSProperties['lineHeight'];
 }
 
 export interface LetterSpacing {
-  xs: React.CSSProperties['letterSpacing'];
-  normal: React.CSSProperties['letterSpacing'];
+  default: React.CSSProperties['letterSpacing'];
+  sm: React.CSSProperties['letterSpacing'];
+  md: React.CSSProperties['letterSpacing'];
   lg: React.CSSProperties['letterSpacing'];
 }
 
-export interface Shadow {
-  ring: React.CSSProperties['boxShadow'];
-  md: React.CSSProperties['boxShadow'];
-}
-
-export interface TypographySystems {
+export interface TypographySystem {
   h1: React.CSSProperties;
   h2: React.CSSProperties;
   h3: React.CSSProperties;
   h4: React.CSSProperties;
   h5: React.CSSProperties;
-  headingSubtitle: React.CSSProperties;
-  body: React.CSSProperties;
-  caption: React.CSSProperties;
-  detail: React.CSSProperties;
-  headingIntro: React.CSSProperties;
-  overline: React.CSSProperties;
-  button: React.CSSProperties;
+  h6: React.CSSProperties;
+  body1: React.CSSProperties;
+  body2: React.CSSProperties;
+  body3: React.CSSProperties;
 }
 
 // ---------------------------------------------------------------
 
-export interface StaticTheme {
+export interface ThemeScales {
   borderRadius: BorderRadius;
-  htmlFontSize: React.CSSProperties['fontSize'];
+  elevation: Elevation;
   fontFamily: FontFamily;
   fontSize: FontSize;
   fontWeight: FontWeight;
   lineHeight: LineHeight;
   letterSpacing: LetterSpacing;
-  typography: TypographySystems;
-  shadow: Shadow;
 }
 
-export interface ThemeWithoutVars extends StaticTheme, ColorSystems {}
-
 /**
  * ==============================================
  * Internal type for definfing default Joy theme.
  * ==============================================
  */
-type BasePaletteRange = 50 | 100 | 200 | 300 | 400 | 500 | 600 | 700 | 800 | 900;
+type BasePaletteRange =
+  | 50
+  | 100
+  | 200
+  | 300
+  | 400
+  | 500
+  | 600
+  | 700
+  | 800
+  | 900
+  | 'textColor'
+  | 'textHoverBg'
+  | 'textActiveBg'
```

Comment:
Is the reason for not adding `textFocusBg` because we are going to use outline?

Reply:
I can't think of any other color that we need

Reply:
The focus style is defined separately (in `theme.focus.default`). Variant only contains `default`, `hover`, `active` and `disabled`.